packages/rs-drive-abci/src/execution/engine/run_block_proposal/v0/mod.rs

```diff
@@ -122,6 +126,85 @@ where
                 .expect("current epoch index should be in range"),
         );
 
+        // If there is a core chain lock update, we should start by verifying it
+        if let Some(core_chain_lock_update) = core_chain_lock_update.as_ref() {
+            if !known_from_us {
+                let verification_result = self.verify_chain_lock(
+                    &block_platform_state,
+                    core_chain_lock_update,
+                    true, // if it's not known from us, then we should try submitting it
+                    platform_version,
+                );
+
+                let VerifyChainLockResult {
+                    chain_lock_signature_is_deserializable,
+                    found_valid_locally,
+                    submitted,
+                    found_valid_by_core,
+                } = match verification_result {
+                    Ok(verification_result) => verification_result,
+                    Err(e) => {
+                        // This will happen if the signature is not
+                        return Ok(ValidationResult::new_with_error(
+                            AbciError::InvalidChainLock(e.to_string()).into(),
+                        ));
+                    }
+                };
+
+                if !chain_lock_signature_is_deserializable {
```

Comment:
Why aren't we returning it as an error from `self.verify_chain_lock` ?

Reply:
In the future we might want to punish proposers for submitting invalid data.